### DISCUSSION TITLE
What is the implementation for email masking

### DISCUSSION THREAD

**Discussion by rakeshSgr** [Sunbird-Lern/Community#41] [Query] [answered]
@amit-tarento @reshmi-nair

We have a few questions regarding the email masking. Please check and let us know

1. How email masking is done in the backend?
2. Is masked email will be stored on DB?
3. If masked email is stored in DB, what is the approach to unmasking the data? 
4. how do you query with a data field that is masked?

cc: @aks30

**Comment by amitec006** [comment, marked as answer]
@rakeshSgr There is no way to unmask the masked identifier.
you can query for these masked identifier like email, phone by providing complete data in search request not just by providing mask data.
